### PR TITLE
kdePackages.calligra: 3.2.1 -> 4.0.1

### LINKED ISSUE #232868: Update request: calligra 3.2.1 → 3.3.0
- Package name: calligra
- Latest released version: 3.3.0

- Current version on the unstable channel: 3.2.1
- Current version on the stable/release channel: 3.2.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@ebzzry 
@zraexy 

* https://calligra.org/news/announcements/stable/calligra-plan-version-3-3-0-released/index.html
* https://download.kde.org/stable/calligra/3.3.0/

-----

Note for maintainers: Please tag this issue in your PR.

### LINKED ISSUE #337989: Update request: calligra 3.2.1 → 4.0.0
- Package name: calligra
- Latest released version: 4.0.0

- Current version on the unstable channel: 3.2.1
- Current version on the stable/release channel: 3.2.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@ebzzry @zraexy

-----
https://invent.kde.org/office/calligra/-/tags/v4.0.0

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by SigmaSquadron**
## Description of changes

Pretty much removes `calligra` and replaces it with a completely new derivation.

Closes #337989.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 344930"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/kde/misc/calligra/default.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    maintainers = with lib.maintainers; [
+      ebzzry
+      zraexy
+      sigmasquadron
+    ];
+    license = with lib.licenses; [
+      gpl2
+      lgpl2
+    ];
+    mainPackage = "calligralauncher";
```
Comment (K900) [inline pkgs/kde/misc/calligra/default.nix:80]: I think you mean `mainProgram`

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -28711,8 +28711,6 @@ with pkgs;
     calicoctl
     confd-calico;
 
-  calligra = libsForQt5.callPackage ../applications/office/calligra { };
```
Comment (K900) [inline pkgs/top-level/all-packages.nix:28714]: Should add an alias to aliases.nix so it doesn't just break people's configs

File: pkgs/kde/misc/calligra/default.nix [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    maintainers = with lib.maintainers; [
+      ebzzry
+      zraexy
+      sigmasquadron
+    ];
+    license = with lib.licenses; [
+      gpl2
+      lgpl2
+    ];
+    mainPrograms = "calligralauncher";
```
Comment (K900) [inline pkgs/kde/misc/calligra/default.nix:80]: `mainProgram`, no `s`.

File: pkgs/kde/misc/calligra/default.nix [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    boost
+    qtkeychain
+    qtsvg
+
+    # Required for Calligra Sheets.
+    eigen
+    imath
+
+    libgit2 # Git integration.
+    okular.dev # Okular integration.
```
Comment (K900) [inline pkgs/kde/misc/calligra/default.nix:44]: Shouldn't need explicit `.dev` on inputs, also comments are a bit too much IMO but not a blocker

File: pkgs/top-level/aliases.nix [hunk 5]
```diff
@@ -321,6 +321,8 @@ mapAliases {
 
   cq-editor = throw "cq-editor has been removed, as it use a dependency that was disabled since python 3.8 and was last updated in 2021"; # Added 2024-05-13
 
+  calligra = kdePackages.calligra; # Added 2024-09-27
```
Comment (K900) [inline pkgs/top-level/aliases.nix:324]: These are roughly alphabetical. Roughly. But still.

File: pkgs/kde/misc/calligra/default.nix [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    perl
+    pkg-config
+  ];
+
+  # Recommended by the upstream packaging instructions. RELEASE_BUILD disables
+  # unmaintained components, like Braindump, from being built, and KDE_NO_DEBUG_OUTPUT
+  # is supposed to improve performance in the finished package.
+  extraCmakeFlags = [
+    "-DRELEASE_BUILD=true"
+    ''-DCMAKE_CXX_FLAGS="-DKDE_NO_DEBUG_OUTPUT"''
```
Comment (K900) [inline pkgs/kde/misc/calligra/default.nix:61]: Is this an upstream recommendation? I feel like debug logs would be pretty useful when dealing with something as not-very-stable as this.

Comment (SigmaSquadron) [reply pkgs/kde/misc/calligra/default.nix:61]: From `README.PACKAGERS`:

> For alpha and beta packages, please build with debug output enabled, but for
production packages the -DCMAKE_CXX_FLAGS="-DKDE_NO_DEBUG_OUTPUT" is recommended.
A significant performance increase will be the result.
>
> All modules may not be ready for production release.
It is recommended to specify -DRELEASE_BUILD=true to exclude these modules.

File: pkgs/top-level/aliases.nix (hunk 5 above)
Comment (SigmaSquadron) [reply pkgs/top-level/aliases.nix:324]: *roughly* is putting in a lot of work there, haha! I added it above `callP...`.

File: pkgs/kde/misc/calligra/default.nix (hunk 6 above)
Comment (SigmaSquadron) [reply pkgs/kde/misc/calligra/default.nix:61]: Resolved on the KDE matrix chat. No overrides.

**Review by Steinhagen — APPROVED** [review]
The code looks good and the application works as expected!

**Comment by SigmaSquadron** [comment]
Result of `nixpkgs-review pr 344930` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>kdePackages.calligra</li>
    <li>kdePackages.calligra.debug</li>
    <li>kdePackages.calligra.dev</li>
    <li>kdePackages.calligra.devtools</li>
  </ul>
</details>

**Pull request merged**